documentation/tutorials/password-change.md

```diff
@@ -0,0 +1,256 @@
+
+This is the `:change_password` action that we are starting with, generated by Igniter.
+```elixir
+# lib/my_app/accounts/user.ex
+    # ...
+    update :change_password do
+      require_atomic? false
+      accept []
+      argument :current_password, :string, sensitive?: true, allow_nil?: false
+
+      argument :password, :string,
+        sensitive?: true,
+        allow_nil?: false,
+        constraints: [min_length: 8]
+
+      argument :password_confirmation, :string, sensitive?: true, allow_nil?: false
+
+      validate confirm(:password, :password_confirmation)
+
+      validate {AshAuthentication.Strategy.Password.PasswordValidation,
+                strategy_name: :password, password_argument: :current_password}
+
+      change {AshAuthentication.Strategy.Password.HashPasswordChange, strategy_name: :password}
+    end
+    # ...
+```
+
+Most web applications that you have used likely had the UI for changing your password somewhere under your personal settings.  We are going to do the same, and create a [LiveComponent](https://hexdocs.pm/phoenix_live_view/Phoenix.LiveComponent.html) to contain our password-change UI, which we can then mount in a LiveView with the rest of the user settings.
+
+Start by defining the module, and defining the template in the `render/1` function.
+```elixir
+# lib/my_app_web/components/change_password_component.ex
+defmodule MyAppWeb.ChangePasswordComponent do
+  @moduledoc """
+  LiveComponent for changing the current user's password.
+  """
+  use MyAppWeb, :live_component
+  alias MyApp.Accounts.User
+
+  @impl true
+  def render(assigns) do
+    ~H"""
+    <div>
+      <.simple_form
+        for={@form}
+        id="user-password-change-form"
+        phx-target={@myself}
+        phx-submit="save"
+      >
+        <.input field={@form[:current_password]} type="password" label="Current Password" />
+        <.input field={@form[:password]} type="password" label="New Password" />
+        <.input field={@form[:password_confirmation]} type="password" label="Confirm New Password" />
+        <:actions>
+          <.button phx-disable-with="Saving...">Save</.button>
+        </:actions>
+      </.simple_form>
+    </div>
+    """
+  end
+end
+```
+
+This will produce a form with the usual three fields (current password, new password, and confirmation of new password, to guard against typographical errors), and a submit button.  Now let's populate the `@form` assign.
+
+```elixir
+  @impl true
+  def update(assigns, socket) do
+    {:ok,
+     socket
+     |> assign(assigns)
+     |> assign_form()}
+  end
+
+  defp assign_form(%{assigns: %{current_user: user}} = socket) do
+    form =
+      AshPhoenix.Form.for_update(user, :change_password,
+        as: "user",
+        prepare_source: fn changeset ->
+          %{
+            changeset
+            | data:
+                Ash.load!(changeset.data, :hashed_password,
```

Comment:
Only necessary if using field policies with private fields set to `:include`. By default private fields are just available.